buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/BuildUtils.kt

```diff
@@ -1,181 +1,186 @@
+    // but don't want the associated tasks to be available; they should only be available in the
+    // root project.
+    tasksAfter.forEach { it.enabled = false }
+    // The only task (as of gradle-intellij-plugin v1.7.0, at least) that needs to be enabled in all
+    // IntelliJ-aware modules
+    // is `classpathIndexCleanup`, to avoid caching issues caused by `classpath.index` file
+    // showing up in build/classes/ and build/resources/ directories.
+    // See https://github.com/JetBrains/gradle-intellij-plugin/issues/1039 for details.
+    project.tasks.withType<ClasspathIndexCleanupTask> { enabled = true }
+
+    project.configure<CheckerFrameworkExtension> {
+      // Technically, UI thread handling errors can happen outside of the (mostly frontend) modules
+      // that depend on IntelliJ,
+      // but the risk is minuscule and not worth the extra computational burden in every single
+      // build.
+      // This might change, however, if/when Checker Framework adds @Heavyweight annotation
+      // (https://github.com/typetools/checker-framework/issues/3253).
+      checkers.add("org.checkerframework.checker.guieffect.GuiEffectChecker")
     }
 
-    fun commonsIO(project: Project) {
-        project.dependencies {
-            val dependencyNotation = "commons-io:commons-io:2.11.0"
-            add("implementation", dependencyNotation)
-        }
+    project.configure<IntelliJPluginExtension> {
+      version.set(IntellijVersionHelper.instance["buildTarget"] as String)
+      // No need to instrument Java classes with nullability assertions, we've got this covered much
+      // better by Checker
+      // (and we don't plan to expose any part of the plugin as an API for other plugins).
+      instrumentCode.set(false)
+      if (withGit4Idea) {
+        plugins.set(listOf("git4idea"))
+      }
     }
+  }
 
-    fun ideProbe(project: Project) {
-        project.repositories {
-            // Needed for com.intellij.remoterobot:remote-robot
-            maven { url = `java.net`.URI("https://packages.jetbrains.team/maven/p/ij/intellij-dependencies") }
-        }
-
-        project.dependencies {
-            add("uiTestImplementation", testFixtures(project(":testCommon")))
-            add("uiTestImplementation","org.virtuslab.ideprobe:junit-driver_2.13:0.36.1")
-            add("uiTestImplementation","org.virtuslab.ideprobe:robot-driver_2.13:0.36.1")
-
-            // This is technically redundant (both since ide-probe pulls in scala-library anyway,
-            // and since ide-probe is meant to use in src/uiTest code, not src/test code),
-            // but apparently needed for IntelliJ to detect Scala SDK version in the project (it's probably https://youtrack.jetbrains.com/issue/SCL-14310).
-            add("testImplementation", "org.scala-lang:scala-library:2.13.8")
-        }
+  fun commonsIO(project: Project) {
+    project.dependencies {
+      val dependencyNotation = "commons-io:commons-io:2.11.0"
```

Comment:
Nit: I'd call a value like that
```suggestion
      val coordinates = "commons-io:commons-io:2.11.0"
```

Reply:
(here and in other places)